barman/backup.py

```diff
@@ -582,6 +582,12 @@ def delete_backup(self, backup, skip_wal_cleanup_if_standalone=True):
                 e.hook.script,
             )
 
+        # Update next rsync backup size information
+        if backup.config.reuse_backup:
+            next_backup = self.get_next_backup(backup.backup_id)
+            if next_backup:
+                self.backup_fsync_and_set_sizes(next_backup)
```

Comment:
We should not rely on volatile information -- the user can change the configuration file.

We should rather rely on the state of the backup, which is stored in its metadata.

Also, simply reusing the `backup_fsync_and_set_sizes` here seems not the best path:

* It will perform a lot of `fsync` syscalls, what puts unnecessary overhead;
* That function has some logging/output which are tied to a backup operation.

I'd say to do something different here, like:

* Create a new method `def _set_backup_sizes(self, backup_info, fsync=False)`
   * It would take care of going through the files
      * If `fsync` then fsync to disk, otherwise simply stat the file
   * Set the attributes in `backup_info` accordingly

That way we could reuse that internal method both when deleting and when backing up.